File: JavascriptAlgorithm/Tree/invertTree-226.js
var invertTree = function (root) {
  if (!root) {
    return null;
  }
  /* let left = invertTree(root.right);
  let right = invertTree(root.left);
  root.left = left;
  root.right = right; */
  let leftTemp = root.left;
  root.left = invertTree(root.right);
  root.right = invertTree(leftTemp);
  return root;
};
export { invertTree };

File: Clion/ZeroTrac/Tree/R1/invertTree_226.cpp
//
// Created by TzashiNorpu on 3/14/2023.
//
#include "tree_r1.h"

using namespace Tree;

TreeNode *Solution::invertTree(TreeNode *root) {
  if (!root) return nullptr;
  TreeNode *l = root->left;
  TreeNode *r = root->right;
  root->left = invertTree(r);
  root->right = invertTree(l);
  return root;
}